.github/workflows/ci-cd.yml

```diff
@@ -46,7 +56,7 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
 
-      - if: steps.conditionals_handler.outputs.is_default_branch == 'true'
+      - if: steps.conditionals_handler.outputs.is_push_to_default_branch == 'true'
```

Comment:
used the newly added `is_push_to_default_branch` condition, instead of the `is_default_branch`, to restrict the new tag generating step to only execute when a push commit to the default branch is made